src/v/cluster/partition.cc

```diff
@@ -1124,45 +1173,133 @@ ss::future<> partition::unsafe_reset_remote_partition_manifest(iobuf buf) {
     // Deserialise provided manifest
     cloud_storage::partition_manifest req_m{
       _raft->ntp(), _raft->log_config().get_initial_revision()};
-    req_m.update_with_json(std::move(buf));
-
-    // A generous timeout of 60 seconds is used as it applies
-    // for the replication multiple batches.
-    auto deadline = ss::lowres_clock::now() + 60s;
-    std::vector<cluster::command_batch_builder> builders;
+    req_m.update_with_json(std::move(json_buf));
 
-    auto reset_builder = _archival_meta_stm->batch_start(deadline, _as);
-
-    // Add command to replace manifest. When applied, the current manifest
-    // will be replaced with the one provided.
-    reset_builder.replace_manifest(req_m.to_iobuf());
+    co_await replicate_unsafe_reset(std::move(req_m));
+}
 
+ss::future<>
+partition::unsafe_reset_remote_partition_manifest_from_cloud(bool force) {
```

Comment:
do we need to ensure mutual exclusion with partition::{start,stop} which may be invoked by the cross core partition movement?

Reply:
I was trying to avoid adding yet another mutex, but I don't think there's a way out of it. Starting and stopping the archiver for updating topic configs was already prone to races. Added a lock. Running the tests locally...

Reply:
I've added a mutex which guards starting and stopping the archiver. I don't really like the approach, but couldn't see how to ensure safety without it.